lib/iris/fileformats/_nc_load_rules/helpers.py

```diff
@@ -1936,6 +1973,38 @@ def is_grid_mapping(engine, cf_name, grid_mapping):
     return is_valid
 
 
+################################################################################
+def _parse_extened_grid_mapping(grid_mapping):
```

Comment:
Looking at the downstream usages: I'm wondering whether the returned object would be better structured as a dictionary of `{coord_name: coord_system_name}`. In the simple case it could be `{None: coord_system_name}`.

Would this make the downstream code easier to write/follow in your opinion? I'm not 100% sure myself.

Reply:
Recoded `_parse_extended_grid_mapping` to return `{coord_name: coord_system}`.
Simplifies the code when building Dim/Aux Coords. Requires a bit of extra processing in `cf.py` when identifying the `CFGridMappingVariable` as we need to iterate over unique coordinate systems.

See bd401c9b9

Reply:
Of course, this looses the ordering of the coordinates in the original mapping...
Not an immediate issue, but potentially a bit of a hurdle if we want to compare- the axis order in the grid mapping against that in the WKT...